openvdb/tree/RootNode.h

```diff
@@ -2587,6 +2591,20 @@ RootNode<ChildT>::addLeafAndCache(LeafNodeType* leaf, AccessorT& acc)
     child->addLeafAndCache(leaf, acc);
 }
 
+template<typename ChildT>
+inline void
+RootNode<ChildT>::addChild(ChildT* child)
+{
+    if (!child) return;
```

Comment:
The behavior of this method is inconsistent with InternalNode::addChild which I find to be confusing. If the incoming pointer is nullptr the method is a noop, whereas InternalNode::addChild assumes this to never happen!

Reply:
Hi Ken,

Thanks for the feedback, I've addressed everything except this comment. I find this situation confusing too, however this inconsistency already exists in other methods across these two classes such as **addLeaf**:

RootNode methods are no-op:
```
RootNode<ChildT>::addLeaf(LeafNodeType* leaf)
{
    if (leaf == nullptr) return;
```

InternalNode methods assume not a nullptr:
```
InternalNode<ChildT, Log2Dim>::addLeaf(LeafNodeType* leaf)
{
    assert(leaf != nullptr);
```

So I had to make a choice here. I don't mind changing one of them if you have a preference, but just highlighting the existing confusion. :/

Reply:
Hi Dan,

Good point - I wonder if now (with a major update) we should change the signature of RootNode::addLeaf? The new Boolean return value should not break existing client code so I don't see the risk - do you?

Reply:
Just to clarify, are you saying that you're ok with the current behavior when passing in a nullptr to all of these addLeaf/addChild methods?

Regarding the return types, there's a somewhat unrelated hole in that LeafNode::addLeaf() does not take ownership of the child, whereas InternalNode and RootNode do. I can address this, but perhaps it's better to do so in a new PR?

Reply:
yes I am okay with the current behavior

Reply:
ok, thanks @kmuseth, I'm going to merge this then.